### PR TITLE
Upgrade major 0x dependencies.

### PR DESCRIPTION
**Pull request by sunwrobert**
Made some minor changes to tslint for compilation reasons.

Upgrades all 0x dependencies to major versions.

Builds and tests pass.

Removed one small assert in the order hash schema validation but I don't think we really need literal asserts in runtime code. If the hash is the same as the one in the test, functionality should be unchanged.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,27 +26,29 @@
...
+    "@0x/abi-gen": "^5.7.0",
+    "@0x/assert": "^3.0.30",
+    "@0x/base-contract": "^6.4.4",
+    "@0x/json-schemas": "^6.4.0",
+    "@0x/utils": "^6.4.4",
+    "@0x/web3-wrapper": "^7.6.1",
     "bn.js": "^4.11.8",
     "ethereumjs-abi": "https://github.com/ProjectWyvern/ethereumjs-abi.git",
     "ethereumjs-util": "^5.1.3",
+    "ethers": "^4.0.49",
```
Comment (Meemaw) [inline package.json:38]: Why is `ethers` needed?

Comment (sunwrobert) [reply package.json:38]: the generated `abi` files use `ethers`. when running `yarn abi_gen`. not sure if there's a better way to convey that it's due to a build artifacts needing it.

File: yarn-error.log [hunk 2]
```diff
@@ -0,0 +1,6307 @@
+Arguments:
```
Comment (Meemaw) [inline yarn-error.log]: This should not be commited, should be in `.gitignore`

**Review by Meemaw — CHANGES_REQUESTED** [review]
`yarn-error.log` should not be commited

**Comment by jameswenzel** [comment]
Some tsc compilation errors (annoying!) but looks great! Can we remove some/all of the `.mustache` files? Not too sure what the new abi-gen is doing but it looks like `contract.mustache` isn't being used, at least.

**Comment by jameswenzel** [comment]
Also, I'm getting warnings from calling `new Buffer` since that's apparently unsafe. In `src/utils/utils.ts` it can be replaced with `Buffer.from` and in `src/wyvernProtocol.ts` with `Buffer.alloc`. Tests still pass for me after those changes

**Comment by sunwrobert** [comment]
> Also, I'm getting warnings from calling `new Buffer` since that's apparently unsafe. In `src/utils/utils.ts` it can be replaced with `Buffer.from` and in `src/wyvernProtocol.ts` with `Buffer.alloc`. Tests still pass for me after those changes

Hmm, I'm not seeing those errors anywhere.

**Comment by jameswenzel** [comment]
> > Also, I'm getting warnings from calling `new Buffer` since that's apparently unsafe. In `src/utils/utils.ts` it can be replaced with `Buffer.from` and in `src/wyvernProtocol.ts` with `Buffer.alloc`. Tests still pass for me after those changes
> 
> Hmm, I'm not seeing those errors anywhere.

Which version of node are you running? I'm on 16.11.0

**Comment by sunwrobert** [comment]
> > > Also, I'm getting warnings from calling `new Buffer` since that's apparently unsafe. In `src/utils/utils.ts` it can be replaced with `Buffer.from` and in `src/wyvernProtocol.ts` with `Buffer.alloc`. Tests still pass for me after those changes
> > 
> > 
> > Hmm, I'm not seeing those errors anywhere.
> 
> Which version of node are you running? I'm on 16.11.0

Gotcha, fixed, had to update @types/node to also be 16.11.0

**Review by Meemaw — APPROVED** [review]

**Pull request merged**